src/main/java/org/opentripplanner/middleware/otp/OtpRequestProcessor.java

```diff
@@ -44,53 +44,55 @@ public static String planning(Request request, spark.Response response) {
         if (isAuthHeaderPresent(request)) {
             Auth0Connection.checkUser(request);
-        }
-
-        Auth0UserProfile profile = Auth0Connection.getUserFromRequest(request);
-
-        if (profile == null || profile.otpUser == null || !profile.otpUser.storeTripHistory) {
-            LOG.debug("Anonymous user or user does not want trip history stored");
-        }
-
-        Response otpResponse = otpDispatcherResponse.getResponse();
-        // only save trip details if the user has given consent and a response from OTP is provided
-        if (profile != null && profile.otpUser.storeTripHistory && otpResponse != null) {
-
-            TripRequest tripRequest = new TripRequest(profile.otpUser.id, batchId, request.queryParams("fromPlace"),
-                request.queryParams("toPlace"), request.queryString());
-
-            TripPlan tripPlan = otpResponse.getPlan();
-
-            TripSummary tripSummary;
-            if (tripPlan != null) {
-                tripSummary = new TripSummary(otpResponse.getPlan().from, otpResponse.getPlan().to,
-                    otpResponse.getError(), otpResponse.getPlan().itinerary, tripRequest.id);
+            Auth0UserProfile profile = Auth0Connection.getUserFromRequest(request);
+            final boolean storeTripHistory = profile != null && profile.otpUser != null && profile.otpUser.storeTripHistory;
+            // only save trip details if the user has given consent and a response from OTP is provided
+            if (!storeTripHistory) {
+                LOG.debug("Anonymous user or user does not want trip history stored");
+            } else if (otpDispatcherResponse.response == null) {
+                LOG.warn("OTP response is null, cannot save trip history for user!");
             } else {
-                tripSummary = new TripSummary(otpResponse.getError(), tripRequest.id);
-            }
-
-            // only save trip summary if the trip request was saved
-            if (saveTripRequest(tripRequest)) {
-                saveTripSummary(tripSummary);
-            } else {
-                LOG.warn("Unable to save trip request, orphaned trip summary not saved");
+                TripRequest tripRequest = new TripRequest(profile.otpUser.id, batchId, request.queryParams("fromPlace"),
+                    request.queryParams("toPlace"), request.queryString());
+                // only save trip summary if the trip request was saved
+                if (saveTripRequest(tripRequest)) {
+                    TripPlan tripPlan = otpDispatcherResponse.response.plan;
+                    TripSummary tripSummary;
+                    if (tripPlan != null) {
+                        tripSummary = new TripSummary(tripPlan, otpDispatcherResponse.response.error, tripRequest.id);
+                    } else {
+                        tripSummary = new TripSummary(otpDispatcherResponse.response.error, tripRequest.id);
+                    }
```

Comment:
I realize that I probably could have added this to #24 (sorry about that), but I think I might prefer if we just handle the tripPlan null check in the TripSummary constructor. That way we can get rid of the overloaded TripSummary constructor (the one with two args).

Reply:
Updated.